### PR TITLE
Implemented ReverseDirectRelationConnection

### PR DESCRIPTION
**Pull request by doctrino**
## Description
Please describe the change you have made.

## Checklist:
- [x] Tests added/updated.
- [x] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [x] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [x] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite/client/data_classes/data_modeling/views.py [hunk 1]
```diff
@@ -392,19 +390,54 @@ def as_apply(self) -> MappedPropertyApply:
...
+    """
+
     type: DirectRelationReference
     source: ViewId
     name: str | None = None
     description: str | None = None
     edge_source: ViewId | None = None
     direction: Literal["outwards", "inwards"] = "outwards"
-    connection_type: Literal["multiEdgeConnection"] = "multiEdgeConnection"
+    connection_type: Literal["single_edge_connection", "multi_edge_connection"] = "multi_edge_connection"
```
Comment (doctrino) [inline cognite/client/data_classes/data_modeling/views.py:440]: The spec has changed, so I changed this to `snake_case` as well

File: cognite/client/data_classes/data_modeling/ids.py [hunk 2]
```diff
@@ -138,6 +144,25 @@ def as_property_ref(self, property: str) -> tuple[str, ...]:
...
 
+@dataclass(frozen=True)
+class PropertyId(CogniteObject):
+    source: ViewId | ContainerId
+    property: str
+
+    @classmethod
+    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+        return cls(
+            source=ViewId.load(resource["source"]),
```
Comment (per-tol) [inline cognite/client/data_classes/data_modeling/ids.py:155]: Shouldn't we check if this is a ContainerId first?

File: cognite/client/data_classes/data_modeling/views.py (hunk 1 above)
Comment (erlendvollset) [inline cognite/client/data_classes/data_modeling/views.py:440]: Let's remove `connection_type` from these classes and just have one class per type connection type instead.

File: cognite/client/data_classes/data_modeling/views.py [hunk 3]
```diff
@@ -446,26 +479,117 @@ def as_apply(self) -> SingleHopConnectionDefinitionApply:
...
+class ReverseSingleHopConnection(ConnectionDefinition):
+    """Describes the direct relation(s) pointing to instances read through this view. This connection type is used to
+    aid in discovery and documentation of the view
+
+    It is called 'ReverseDirectRelationConnection' in the API spec.
+
+    Args:
+        source (ViewId): The node(s) containing the direct relation property can be read through
+            the view specified in 'source'.
+        through (PropertyId): The view or container of the node containing the direct relation property.
```
Comment (erlendvollset) [inline cognite/client/data_classes/data_modeling/views.py:496]: `NodeResultExpression.through` is a `list[str] | tuple[str, str, str]`. Let's just do that here aswell to stay consistent.

File: cognite/client/data_classes/data_modeling/ids.py (hunk 2 above)
Comment (per-tol) [reply cognite/client/data_classes/data_modeling/ids.py]: Fixed and added test

File: cognite/client/data_classes/data_modeling/views.py (hunk 1 above)
Comment (per-tol) [reply cognite/client/data_classes/data_modeling/views.py]: done

File: cognite/client/data_classes/data_modeling/views.py (hunk 3 above)
Comment (per-tol) [reply cognite/client/data_classes/data_modeling/views.py:524]: Do you mean we'd use the array length to reason if we're dealing with a view or container reference, or reusing the logic in DMS to reference a view as [space, externalId/version, property]? (compared with [space, externalId, version, property]? Personally I never liked that check if id[1] contains '/'.

Comment (erlendvollset) [reply cognite/client/data_classes/data_modeling/views.py:524]: View and container references have the same length - views are (space, extid/version, prop) while for containers it's (space, extid, prop). 

But it's to be consistent within the SDK, and property references are lists elsewhere. Accepting both forms is fine with me though.

File: tests/tests_unit/test_data_classes/test_data_models/test_views.py [hunk 4]
```diff
@@ -52,19 +52,62 @@ def test_load_dumped_mapped_property_for_apply(self) -> None:
             "source": {"space": "mySpace", "external_id": "myExternalId", "version": "myVersion", "type": "view"},
         }
 
-    def test_load_dump_connection_property(self) -> None:
+    def test_load_dump_single_edge_connection_property(self) -> None:
```
Comment (erlendvollset) [inline tests/tests_unit/test_data_classes/test_data_models/test_views.py:55]: load/dump roundtrip tests are actually generated for subclasses of CognitrObject, so these are redundant.

File: tests/tests_integration/test_api/test_data_modeling/test_views.py [hunk 5]
```diff
@@ -308,3 +317,31 @@ def test_apply_different_property_types(
...
+                "persons": SingleReverseDirectRelationApply(
+                    source=person_view.as_id(),
+                    name="Person",
+                    through=PropertyId(source=actor_view.as_id(), property="person"),
+                )
+            },
+        )
+
+        try:
+            # Act
```
Comment (erlendvollset) [inline tests/tests_integration/test_api/test_data_modeling/test_views.py:340]: Let's drop the Act/Assert/Cleanup comments.

File: tests/tests_unit/test_data_classes/test_data_models/test_views.py (hunk 4 above)
Comment (per-tol) [reply tests/tests_unit/test_data_classes/test_data_models/test_views.py]: removed tests using the default behaviour covered by the generated tests.

**Review by erlendvollset — APPROVED** [review]
Added a small encapsulation fix. LGTM!

**Pull request merged**